### ISSUE TITLE
Dynamic configs/credentials (depending on end-user)

### ISSUE THREAD

**Issue by avonian** [connery-io/connery#107]
Basically, I was curious to know how we can pass user-specific secrets to the plugin, since I'm looking to create an agent that would service multiple users for the same kinds of functions (gmail being a great example).

**Comment by machulav** [comment]
Hi @avonian,

This is a very good question! It's not documented yet, but we support this already. You can add additional parameter into the

**Comment by machulav** [comment]
Hi @avonian, this is a good question! It still needs to be documented, but we support this already. You can add additional property `configuration` into the body when running the action like the following:

```
curl -X 'POST' \
  'http://localhost:4201/api/actions/sendEmail/run' \
  -H 'accept: application/json' \
  -H 'x-api-key: 123456' \
  -H 'Content-Type: application/json' \
  -d '{
  "input": {
    "recipient": "",
    "subject": "",
    "body": ""
  },
  "configuration": {
    "gmailEmailAddress": "",
    "gmailAppPassword": "",
    "senderName": ""
  }
}'
```

In this case, the provided configuration will override the configuration specified in the ENV variables.

**Comment by avonian** [comment]
Wonderful, thank you.

I actually haven't worked with openai agents directly yet (have been created my own "agents" with custom prompt chains), and I figured maybe I was misunderstanding something, which is the reason I erased the original contents of my post, but I've reincorporated them now.

**Comment by machulav** [comment]
You're completely right with your comments! Regarding your original question about authorization, in the future, plugins will also handle authentication for third-party services, so it would be much easier for the agent builders to handle the user's authentication for their services, like Gmail. So every user can connect the plugin to their own account.

P.S.: I reopened the issue immediately after you closed it because I thought that I closed it unintentionally - haha 🙂
